distro/run/core/src/main/java/org/camunda/bpm/run/CamundaBpmRunProcessEngineConfiguration.java

```diff
@@ -16,38 +16,41 @@
  */
 package org.camunda.bpm.run;
 
-import jakarta.inject.Inject;
-import java.io.File;
-import java.io.IOException;
-import java.util.Set;
-
 import org.camunda.bpm.engine.ProcessEngineException;
+import org.camunda.bpm.engine.impl.cfg.CompositeProcessEnginePlugin;
+import org.camunda.bpm.engine.impl.cfg.ProcessEnginePlugin;
 import org.camunda.bpm.engine.impl.diagnostics.CamundaIntegration;
 import org.camunda.bpm.engine.spring.SpringProcessEngineConfiguration;
-import org.springframework.core.env.Environment;
+import org.camunda.bpm.run.property.CamundaBpmRunProcessEnginePluginProperty;
+import org.camunda.bpm.run.property.CamundaBpmRunProperties;
+import org.camunda.bpm.run.utils.CamundaBpmRunProcessEnginePluginHelper;
 import org.springframework.core.io.Resource;
 
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
 public class CamundaBpmRunProcessEngineConfiguration extends SpringProcessEngineConfiguration {
 
+  private final String normalizedDeploymentDir;
 
-  @Inject
-  private Environment env;
+  public CamundaBpmRunProcessEngineConfiguration(CamundaBpmRunProperties runProperties,
```

Comment:
From a cohesion point of view, `CamundaBpmRunProcessEngineConfiguration` (its a engine config, not a spring config) having direct reference in `runProperties` is a direct coupling which doesn't look right to me.

Externalising all dependencies in `CamundaBpmRunConfiguration` fits wells with the Dependency Injection principle.